hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -1589,8 +1580,13 @@ protected Option<HoodieBaseFile> fetchLatestBaseFile(String partitionPath, Strin
    * @return File Slice if present
    */
   public Option<FileSlice> fetchLatestFileSlice(String partitionPath, String fileId) {
-    return Option
-        .fromJavaOptional(fetchLatestFileSlices(partitionPath).filter(fs -> fs.getFileId().equals(fileId)).findFirst());
+    try {
```

Comment:
the invoker is already guarded by a write lock

Reply:
`HoodieCDCExtractor` directly calls this method with a write lock. It is also a read so will need a read lock when reading from the store.